packages/stitch/tests/mergeDeepInterfaces.test.ts

```diff
@@ -1,458 +1,470 @@
+
+  type CategoryDecorator implements TransactionDecorator {
+    id: ID!
+    decorationKey: String!
+    decorationType: DecorationType!
+  }
+
+  type DocumentDecorator implements TransactionDecorator {
+    id: ID!
+    decorationKey: String!
+    decorationType: DecorationType!
+  }
+
+  type DemoType {
+    id: ID!
+    name: String
+  }
+
+  type Review {
+    id: ID!
+    body: String!
+    # product: Product
+    user: User
+    transaction: Transaction
+  }
+
+  type User {
+    id: ID!
+    reviews: [Review]!
+  }
+
+  type Query {
+    reviews(ids: [ID!]!): [Review]!
+    _users(ids: [ID!]!): [User]!
+    _transactions(ids: [ID!]!): [Transaction]!
+    _decoratedTransaction(ids: [ID!]!): [Transaction]!
+    demoTypesByIds(ids: [ID!]!): [DemoType]!
+    categoryDecoratorsByIds(ids: [ID!]!): [CategoryDecorator]!
+    documentDecoratorsByIds(ids: [ID!]!): [DocumentDecorator]!
+    transactionDecoratorsById(id: ID!): TransactionDecorator
+    transactionDecoratorsByIds(ids: [ID!]!): [TransactionDecorator]!
+    _decorators(ids: [ID!]!): [TransactionDecorator]!
+  }
+`;
 const decorators = [
-    { id: '1', txId: 'tx1', decorationKey: 'dtx1', decorationType: 'CategoryDecorator', __typename: 'CategoryDecorator' },
-    { id: '2', txId: 'tx2', decorationKey: 'dtx2', decorationType: 'CategoryDecorator', __typename: 'CategoryDecorator' },
-    { id: '3', txId: 'tx3', decorationKey: 'dtx3', decorationType: 'DocumentDecorator', __typename: 'DocumentDecorator' },
-    { id: '4', txId: 'tx1', decorationKey: 'dtx4', decorationType: 'DocumentDecorator', __typename: 'DocumentDecorator'  },
+  {
+    id: '1',
+    txId: 'tx1',
+    decorationKey: 'dtx1',
+    decorationType: 'CategoryDecorator',
+    __typename: 'CategoryDecorator',
+  },
+  {
+    id: '2',
+    txId: 'tx2',
+    decorationKey: 'dtx2',
+    decorationType: 'CategoryDecorator',
+    __typename: 'CategoryDecorator',
+  },
+  {
+    id: '3',
+    txId: 'tx3',
+    decorationKey: 'dtx3',
+    decorationType: 'DocumentDecorator',
+    __typename: 'DocumentDecorator',
+  },
+  {
+    id: '4',
+    txId: 'tx1',
+    decorationKey: 'dtx4',
+    decorationType: 'DocumentDecorator',
+    __typename: 'DocumentDecorator',
+  },
 ];
 
+const reviews = [{ id: 'tx1', productUpc: '1', userId: '1', body: 'love it' }];
 
-const reviews = [
-    { id: 'tx1', productUpc: '1', userId: '1', body: 'love it' },
-];
-
-const localResolvers = {
-    Transaction: {
-        decorators: (_, { ids }) => {
-            console.log('Transaction decorators');
-            const result = ids.map(id => decorators.find(d => d.txId === id) || null);
-            console.log('CardTransaction decorators', result);
-            return result;
-        },
+const localResolvers: IResolvers = {
+  Transaction: {
+    decorators: ({ id }: { id: string }) => {
+      console.log('Transaction decorators');
+      return decorators.filter(d => d.txId === id);
     },
-    CardTransaction: {
-        decorators: (_, { ids }) => {
```

Comment:
This was wrong because `decorators` don't have an argument called `ids` but it needs `id` from the parent `id`